### PR TITLE
Payment subscription status: Ibex listener + publish event

### PR DESCRIPTION
**Pull request by brh28**

### CODE REVIEW

File: src/services/ledger/index.ts [hunk 1]
```diff
@@ -42,6 +42,10 @@ import { volume } from "./volume"
 import Ibex from "@services/ibex"
 import { IbexApiError, IbexAuthenticationError, IbexEventError } from "@services/ibex/errors"
 
+// FLASH FORK: import ibex dependencies
+import Ibex from "@services/ibex"
```
Comment (brh28) [inline src/services/ledger/index.ts:46]: Unused imports?

File: src/graphql/public/root/subscription/ln-invoice-payment-status.ts [hunk 2]
```diff
@@ -50,6 +50,8 @@ const LnInvoicePaymentStatusSubscription = {
   },
 
   subscribe: async (_source: unknown, args: LnInvoicePaymentSubscribeArgs) => {
+    baseLogger.info("Running subscribe: ln-invoice-payment-status")
```
Comment (brh28) [inline src/graphql/public/root/subscription/ln-invoice-payment-status.ts:53]: Leaving for now. Can be used for integration testing, but can eventually be removed

File: src/servers/trigger.ts [hunk 3]
```diff
@@ -479,26 +480,40 @@ const listenerBria = async () => {
...
+  // const callback = (event) => baseLogger.info(event, "Ibex event received")
+  // const ibexEventHandler = wrapAsyncToRunInSpan({
+  //   namespace: "servers.trigger",
+  //   fn: callback // (event: IbexEvent) => baseLogger.info(event, "Ibex event received"),
+  // })
+}
+
 const main = () => {
-  listenerBria()
+  logger.info("Setting subscribers:")
```
Comment (brh28) [inline src/servers/trigger.ts:495]: remove

File: src/services/notifications/index.ts [hunk 4]
```diff
@@ -16,11 +16,78 @@ import {
...
+      const psResp = await pubsub.publish({
+        trigger: lnPaymentStatusTrigger,
+        payload: { status: "PAID" },
+      })
+      if (psResp instanceof PubSubServiceError) {
+        log.error("Failed to publish")
+        return new NotificationsServiceError(psResp) 
+      }
+      log.info(lnPaymentStatusTrigger, "Message published.")
+      // Notify the recipient (via GraphQL subscription if any)
```
Comment (brh28) [inline src/services/notifications/index.ts:44]: remove commented code

File: src/services/ibex/authenticated-client.ts [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        logRequest("generateBitcoinAddress", body)
+        return withAuth(() => IbexSDK.generateBitcoinAddress(body))
+            .catch(_ => new IbexApiError(_.status, _.data))
+            .then(logResponse) 
+    }
+
+    async addInvoice(body: types.AddInvoiceBodyParam): Promise<types.AddInvoiceResponse201 | IbexAuthenticationError | IbexApiError> {
+        const bodyWithHooks = { 
+            ...body,
+            webhookUrl: 'http://localhost:8889/invoice/receive/status',
```
Comment (brh28) [inline src/services/ibex/authenticated-client.ts:57]: get from environment

File: src/services/ibex/authenticated-client.ts [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        logRequest("getFeeEstimation", metadata)
+        return withAuth(() => IbexSDK.getFeeEstimation(metadata))
+            .catch(_ => new IbexApiError(_.status, _.data))
+            .then(logResponse)
+    }
+
+    async payInvoiceV2(body: types.PayInvoiceV2BodyParam): Promise<types.PayInvoiceV2Response200 | IbexAuthenticationError | IbexApiError> {
+        const bodyWithHooks = { 
+            ...body,
+            webhookUrl: 'http://localhost:8889/invoice/pay/status',
```
Comment (brh28) [inline src/services/ibex/authenticated-client.ts:85]: get from environment

File: src/services/ledger/index.ts (hunk 1 above)
Comment (islandbitcoin) [reply src/services/ledger/index.ts]: remove them if not needed

File: src/servers/trigger.ts (hunk 3 above)
Comment (islandbitcoin) [reply src/servers/trigger.ts]: does this mean if we decide in the future to use the internal wallet, we will not be able to use Bria?

File: .env [hunk 7]
```diff
@@ -120,4 +120,5 @@ export KRATOS_ADMIN_API="http://localhost:4434"
 export IBEX_URL="https://api-sandbox.poweredbyibex.io"
 export IBEX_EMAIL="mail2michaelennis@gmail.com"
 export IBEX_LISTENER_HOST=${DOCKER_HOST_IP}
-export IBEX_LISTENER_PORT=50053
\ No newline at end of file
+export IBEX_LISTENER_PORT=50053
+export IBEX_WEBHOOK_SECRET="not-so-secret"
```
Comment (islandbitcoin) [inline .env:124]: here we definitely need to follow the same method used by upstream to store secrets

Comment (islandbitcoin) [reply .env:124]: .env.local ?

**Review by islandbitcoin — APPROVED** [review]
ACK

**Pull request merged**